pkg/controller/nodes/predicate.go

```diff
@@ -66,6 +68,8 @@ func CanExecute(ctx context.Context, dag executors.DAGStructure, nl executors.No
 				return PredicatePhaseUndefined, errors.Errorf(errors.BadSpecificationError, nodeID, "Upstream node [%v] of node [%v] not defined", upstreamNodeID, nodeID)
 			}
 
+			// Deprecated: This if block will be removed in a future version. It's harmless (will be no-op) for newly
```

Comment:
Good catch!
But question, is it really harmless? 
So we will have ParentNodeID set, but branch node will not be an upstream. If so yes, this is harmless